File: MyFrame/mvpframe/src/main/java/zs/xmx/view/activity/RegisterActivity.java
package zs.xmx.view.activity;

import android.util.Log;
import android.view.View;

import javax.inject.Inject;

import zs.xmx.R;
import zs.xmx.base.BaseActivity;
import zs.xmx.dagger2.component.DaggerRegisterComponent;
import zs.xmx.dagger2.module.RegisterPresenterModule;
import zs.xmx.net.Rx.databus.RxBus;
import zs.xmx.presenter.RegisterPresenter;
import zs.xmx.view.IRegisterView;

public class RegisterActivity extends BaseActivity implements IRegisterView {

    @Inject
    RegisterPresenter mPresenter;

    @Override
    protected int setContentViewId() {
        return R.layout.activity_register;
    }

    @Override
    protected void initView() {
        findViewById(R.id.button).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button:
                break;
        }
    }

    @Override
    protected void initEvent() {
        //new RegisterPresenter(this).fetch();
        //RxBus注册(类似EventBus)
        //mPresenter = new RegisterPresenter(this);
        DaggerRegisterComponent.builder()
                .registerPresenterModule(new RegisterPresenterModule(this))
                .build().inject(this);
        RxBus.getInstance().register(mPresenter);
    }


    @Override
    public void registerData(String datas) {
        Log.i(TAG, "registerData: " + datas);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        //RxBus取消注册(类似EventBus)
        RxBus.getInstance().unRegister(mPresenter);
    }
}

File: MyFrame/mvpframe/src/main/java/zs/xmx/mvpframe/ui/activity/RegisterActivity.kt
package zs.xmx.mvpframe.ui.activity

import android.view.View
import kotlinx.android.synthetic.main.activity_register.*
import zs.xmx.mvpframe.R
import zs.xmx.mvpframe.base.activity.BaseMvpActivity
import zs.xmx.mvpframe.constant.MyConstant
import zs.xmx.mvpframe.ext.onClick
import zs.xmx.mvpframe.injection.component.DaggerUserComponent
import zs.xmx.mvpframe.injection.module.UserModule
import zs.xmx.mvpframe.presenter.RegisterPresenter
import zs.xmx.mvpframe.utils.Logger
import zs.xmx.mvpframe.view.impl.activity.IRegisterView
import java.util.*

class RegisterActivity : BaseMvpActivity<RegisterPresenter>(), IRegisterView {

    override fun initComponentInject() {
        DaggerUserComponent.builder()
                .activityComponent(mActivityComponent)
                .userModule(UserModule())
                .build()
                .inject(this)
        mPresenter.mView = this
    }


    override fun setContentViewId(): Int {
        return R.layout.activity_register
    }


    override fun initEvent() {
        btn_regiest.onClick(this)
        mVerifyCodeBtn.onClick(this)
    }


    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_regiest -> {
                val params = HashMap<String, Any>()
                params["key"] = MyConstant.MOB_KEY
                params["username"] = et_userName.text.toString()
                params["password"] = et_pwd.text.toString()
                mPresenter.register(MyConstant.REGISTER, params, TAG)
            }
            R.id.mVerifyCodeBtn->{
                mVerifyCodeBtn.requestSendVerifyNumber()
            }
        }
    }


    override fun onRegisterResult(result: Any) {
        Logger.e(TAG, result.toString())
        showToast(result.toString())
    }

    override fun onDestroy() {
        super.onDestroy()
        //RXBus反注册
        mPresenter.unRegisterBus()
    }


}
